src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/AssetService.cs

```diff
@@ -108,7 +108,7 @@ public bool AttachMeshFilter(GameObject gameObject, Guid assetId)
             ComponentExtensions.EnsureComponent<MeshRenderer>(gameObject);
 
             Mesh mesh = meshAssets.GetAsset(assetId);
-            if (mesh != null)
+            if ((mesh != null) || (assetId == Guid.Empty))

```

Comment:
>if ((mesh != null) || (assetId == Guid.Empty)) [](start = 12, length = 46)

So if the assetId is a Guid.Empty, we don't have to check the cache, and furthermore, if we aren't treating that as an error when sending across, how do we handle it here, do we still attach a meshfilter, do we skip attaching but return true, or do we return false?

Reply:
The intention here was to treat "Guid.Empty" to mean "the null asset".

I was on the fence though about whether I should still let the call to GetAsset happen even when assetId is Guid.Empty or just short-circuit it to null here.

In the end, I decided to let the GetAsset call happen, because I thought perhaps in the future, Guid.Empty and the null asset might be treated in a special way, but I'm open to short-circuiting it here instead.

Since sync'ing across a null mesh is now a supported scenario, I figured it should return true.

---
In reply to: [337248018](https://github.com/microsoft/MixedReality-SpectatorView/pull/250#discussion_r337248018) [](ancestors = 337248018)

Reply:
As discussed elsewhere, I reverted this change, and we can consider it in another PR.